src/scripts/background.js

```diff
@@ -286,7 +286,13 @@ var TogglButton = {
   },
 
   updateCurrentEntry: function (data) {
-    var entry = data.data;
+    var entry = data.data,
+      defaultProject = Db.getDefaultProject(),
+      rememberProjectPer = Db.get("rememberProjectPer");
+    if (rememberProjectPer) {
+      Db.setDefaultProject(entry.pid === defaultProject ? null : entry.pid,
```

Comment:
This bit could use a comment. I went back and forth 6 times trying to figure out how this line works and doesn't mess up the global default project.